### PR TITLE
Fix cpplint warns and add TTD performance checks.

### PR DESCRIPTION
**Pull request by mrkmarron**
- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] documentation is changed or added

##### Affected core subsystem(s)

chakrashim

##### Description of change
Fix various cpplint warnings (Issue #196).
Add fast checks for "TTD Enabled" around TTD buffer modification calls (Issue #193).

PR-URL: https://github.com/mrkmarron/node-chakracore/commit/5d45e13ad4c8203b9eb5edac0c94e86d2edf18fd

### CODE REVIEW

**Comment by kunalspathak** [comment]
LGTM, if CI is green. 
https://ci.nodejs.org/job/chakracore-test-windows/51/

File: deps/chakrashim/src/v8arraybuffer.cc [hunk 1]
```diff
@@ -107,20 +107,21 @@ ArrayBuffer::Contents ArrayBuffer::GetContents() {
 }
 
 // ENABLE_TTD
-void ArrayBuffer::TTDRawBufferNotifyRegisterForModification(byte* initialModPosition) {
-    JsTTDRawBufferAsyncModificationRegister(this, initialModPosition);
+void ArrayBuffer::TTDRawBufferNotifyRegisterForModification(
+  byte* initialModPosition) {
```
Comment (kfarnung) [inline deps/chakrashim/src/v8arraybuffer.cc:111]: In other areas I see two indents when the parameters start on the following line.  This helps keep them from blending in with the body of the function.

File: src/node.cc [hunk 2]
```diff
@@ -3793,38 +3800,38 @@ static void ParseArgs(int* argc,
       trace_enabled_categories = categories;
 #if ENABLE_TTD_NODE
       // Parse and extract the TT args
-      } else if(strcmp(arg, "--record") == 0) {
+      } else if (strcmp(arg, "--record") == 0) {
           s_doTTRecord = true;
```
Comment (kfarnung) [inline src/node.cc:3804]: Shouldn't everything in this file use an indent of two spaces?

File: src/node.cc [hunk 3]
```diff
@@ -4667,7 +4674,7 @@ inline int Start(uv_loop_t* event_loop,
 #endif
 
 #if ENABLE_TTD_NODE
-  if(s_doTTRecord) {
+  if (s_doTTRecord) {
       fprintf(stderr, "Recording is enabled (but not yet started)...\n");
```
Comment (kfarnung) [inline src/node.cc:4678]: Two space indentation

File: src/node.cc [hunk 4]
```diff
@@ -4929,28 +4936,29 @@ int Start(int argc, char** argv) {
   v8_initialized = true;
 
 #if ENABLE_TTD_NODE
-  bool chk_debug_enabled = debug_options.debugger_enabled() || debug_options.inspector_enabled();
+  bool chk_debug_enabled = debug_options.debugger_enabled()
+                           || debug_options.inspector_enabled();
```
Comment (kfarnung) [inline src/node.cc:4940]: I don't remember why I think this, but I thought there was a standard somewhere to place the operator at the end of the last line.  If that isn't the case, just ignore me.

File: src/node_buffer.h [hunk 5]
```diff
@@ -50,8 +50,10 @@ NODE_EXTERN void TTDSyncDataModNotify(v8::Local<v8::Object> val,
 // Notify us that a native buffer access (which we don't currently
 // understand/support) happened.
 #define TTD_NATIVE_BUFFER_ACCESS_NOTIFY(X) \
-    JsTTDCheckAndAssertIfTTDRunning( \
-            "Unsupported raw buffer access -- investigate this!!!\n")
+    if (s_doTTRecord || s_doTTReplay) { \
+      JsTTDCheckAndAssertIfTTDRunning( \
+            "Unsupported raw buffer access -- investigate this!!!\n"); \
```
Comment (kfarnung) [inline src/node_buffer.h:55]: Shouldn't this have four spaces of indentation?

**Comment by kfarnung** [comment]
A few nit comments, otherwise it LGTM 🚢

**Comment by kfarnung** [comment]
Landed as 25ca0d9eb8e2f9b3522dbd16248aa19e3e231a68

**Pull request closed without merging**